File: submissions/fire emblem/problem1.java
import java.io.*;
import java.util.*;
class problem1
{public static void main(String args[])
{Scanner sc=new Scanner(System.in);
int a=sc.nextInt();
int t=a;
int f,s,k;
s=0;
while(t>0)
{f=t%10;
k=problem1.fact(f);
s=s+k;
t=t/10;
}
if(s==a)
{System.out.println("Yes");
}
else
{System.out.println("No");
}
}
public static int fact(int l)
{int f=1;
for(int i=1;i<=l;i++)
{f=f*i;
}
return f;
}
}

File: submissions/Dynamics/problem1.cpp
#include<stdio.h>
int main()
{
	int d,n=0,i,s=0,m=0,fact=1;
	scanf("%d",&n);
	m=n;
	while(n!=0)
	{
		d=n%10;
		
		for(i=1;i<=d;i++)
		{
			fact=fact*i;
		}
		s=s+fact;
		n=n/10;
	}
	if(m==s)
	{
		printf("Yes");
	}
	else
	{
	printf("No");
	}
	return 0;
}
